homeassistant/components/onvif/event.py

```diff
@@ -174,11 +174,20 @@ async def async_parse_messages(self, messages) -> None:
                     UNHANDLED_TOPICS.add(topic)
                 continue
 
-            event = await parser(unique_id, msg)
+            try:
+                event = await parser(unique_id, msg)
```

Comment:
It doesn't look like there is a case where the parser can return `None` now. Can we change the type here and move the logger into the exception handling block?

Reply:
`async_parse_tplink_detector` can still return `None`.  So we can leave it or another option might be to define a new exception type in `parsers.py` that parsers are allowed to raise. WDYT?

Reply:
I missed that one. 

I think its fine as-is